shared/lib_irradproc.cpp

```diff
@@ -1090,12 +1106,25 @@ int irrad::calc()
 	double t_sunrise = sunAnglesRadians[4];
 	double t_sunset = sunAnglesRadians[5];
 
+	if (t_sunset > 24) //sunset is legitimately the next day, so recalculate sunset from the previous day
+	{
+		double sunanglestemp[9];
+		solarpos(year, month, day - 1, 12, 0.0, latitudeDegrees, longitudeDegrees, timezone, sunanglestemp);
+		t_sunset = sunanglestemp[5] - 24.0;
+	}
+
+	if (t_sunrise < 0) //sunrise is legitimately the previous day, so recalculate for next day
+	{
+		double sunanglestemp[9];
+		solarpos(year, month, day + 1, 12, 0.0, latitudeDegrees, longitudeDegrees, timezone, sunanglestemp);
+		t_sunset = sunanglestemp[5] + 24.0;
```

Comment:
Before adding or subtracting 24, I would check that it is indeed greater than or less than 24, unless we're at that unlucky day where yesterday sunset was at 11:59 pm and today sunset is tomorrow at 12:01 am.

Reply:
Good point on both of these pieces, updating the code accordingly.